client/src/pages/EducatorDashboard/EducatorDashboard.tsx

```diff
@@ -0,0 +1,279 @@
+    tabs: {
+        backgroundColor: theme.palette.primary.light,
+        boxShadow: '0',
+        paddingTop: '2em'
+    },
+    dateFilter: {
+        paddingTop: "2em",
+        display: "flex",
+        flexDirection: "row"
+    },
+    dateFilterText: {
+        padding: "4px 0px"
+    },
+    dateFilterBoxes: {
+        padding: "0px 10px"
+    }
+
+}));
+
+// TAB PANEL CODE FROM MATERIAL UI - may want to reuse this or create a global tab component
+
+interface TabPanelProps {
+    children?: React.ReactNode;
+    index: any;
+    value: any;
+}
+
+function TabPanel(props: TabPanelProps) {
+    const { children, value, index, ...other } = props;
+
+    return (
+        <div
+            role="tabpanel"
+            hidden={value !== index}
+            id={`simple-tabpanel-${index}`}
+            aria-labelledby={`simple-tab-${index}`}
+            {...other}
+        >
+            {value === index && (
+                <Container >
+                    <Box >
+                        {children}
+                    </Box>
+                </Container>
+            )}
+        </div>
+    );
+}
+
+function a11yProps(index: any) {
+    return {
+        id: `simple-tab-${index}`,
+        'aria-controls': `simple-tabpanel-${index}`,
+    };
+}
+
+const EducatorDashboard: React.SFC<Props> = ({ events, fetchEvents, changeFilter }: Props) => {
+    const classes = useStyles();
+
+    //State variables for educator dashboard
+    const [startDate, setStartDate] = useState<Date | null>(null);
+    const [endDate, setEndDate] = useState<Date | null>(null);
+    const [isPastEvent, setIsPastEvent] = useState(false)
+    const [retrievedData, setRetrievedData] = useState(false);
+    const [tabValue, setTabValue] = useState(0)
+
+    // State variables for infinite scroll functionality
+    const [page, setPage] = useState<number>(0);
+    const [prevY, setPrevY] = useState<number>(0);
+    const [lastEventListLength, setLastEventListLength] = useState<number>(0);
+    const [loadedAllEvents, setLoadedAllEvents] = useState<boolean>(false);
+    const offset = 10;
+
+    const loadingRef = useRef() as React.MutableRefObject<HTMLInputElement>
+
+    const handleObserver = useCallback((entities: any, observer: IntersectionObserver) => {
+        const y = entities[0].boundingClientRect.y;
+        const newPage = page + 1;
+
+        if (prevY > y) {
+            if (lastEventListLength === events.length) {
+                console.log("no new events are available")
+                // If no new events are available, prevent additional calls to backend.
+                setLoadedAllEvents(true);
+            }
+
+            if (!loadedAllEvents) {
+                if (events.length > 1)
+                    setLastEventListLength(events.length);
+
+                fetchEvents(offset, offset * newPage);
+                setPage(newPage);
+            }
+        }
+        setPrevY(y)
+
+    }, [page, prevY, lastEventListLength, loadedAllEvents])
```

Comment:
looks like this hook and the hook on line 166 have some missing dependencies.

Reply:
Resolved with [3ffd489](https://github.com/uwblueprint/BEP/pull/63/commits/3ffd4898dcb82a272c037384b6455d37cdaaaedf)